src/Dynode/model/core/socket/socket.ts

```diff
@@ -1,109 +1,109 @@
 import { VEvent, VEventTarget } from 'src/utils/vanillaEvent';
-import { JsonValue } from 'src/utils/objectUtils';
-import ValueType from './valueType';
+import { Value, ValueConstructor } from './value';
 
 /**
  * Class for dealing with Node values.
  */
-class Socket extends VEventTarget {
+class Socket<T> extends VEventTarget {
   /**
-   * The ValueType name of the socket.
+   * ValueConstructor that is used to determine the type of value.
    */
-  protected typeObject: ValueType | null = null;
+  protected ValueType: ValueConstructor<T>;
 
   /**
-   * The stored value.
+   * The Value of the socket. null means there is nothing.
    */
-  value: unknown;
+  protected value: Value<T> | null;
 
   /**
-   * Denotes whether the stored value is equal to nothing.
-   */
-  nothing = false;
-
-  /**
-   * Denotes whether the object holds a value.
+   * Denotes whether the object's value has been set.
+   *
+   * Although it may seem that when value = null the value has not been set, that is not
+   * the case. When value = null and isSetVariable = true it's an explicit signal that there
+   * is no value.
    */
-  hasValue = false;
+  private isSetVariable: boolean;
 
   /**
    * Identifier used to retrieve the socket.
    */
-  name: string | null = null ;
+  public name: string | null = null ;
 
   /**
    * A name used for display.
    */
-  title: string | null = null;
+  public title: string | null = null;
 
-  constructor(typeObject?: ValueType) {
+  constructor(value?: T, ValueType?: ValueConstructor<T>) {
```

Comment:
I think, we don't need to pass the socket value at construction time. Normally, the value is set when the network is resolved. Also, if a value is set at construction time, nobody can subscribe to the socket's "value" event to catch the value update.

Reply:
Agreed, I'll remove value from the constructor.